db/db_iter.cc

```diff
@@ -424,40 +433,66 @@ void DBIter::FindNextUserEntryInternal(bool skipping, bool prefix_check) {
           case kTypeSingleDeletion:
             // Arrange to skip all upcoming entries for this key since
             // they are hidden by this deletion.
-            saved_key_.SetUserKey(
+            // if iterartor specified start_seqnum we
+            // 1) return internal key, including the type
+            // 2) return ikey only if ikey.seqnum >= start_seqnum_
+            // not that if deletion seqnum is < start_seqnum_ we
+            // just skip it like in normal iterator.
+            if (start_seqnum_ > 0 && ikey_.sequence > start_seqnum_)  {
+              saved_key_.SetInternalKey(ikey_);
+              skipping = true;
```

Comment:
I guess setting skipping isn't meaningful since it returns on the next line

Reply:
@ajkr i'm unsure if we need to set `num_skipped = 0;` before we do `return;`.  I think we don't?